### DISCUSSION TITLE
I want to render blade component in column closure.

### DISCUSSION THREAD

**Discussion by tahaRazaTalibi** [Power-Components/livewire-powergrid#1054] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary

I would like to request the ability to render Blade components within column closures in Livewire's PowerGrid.

### Why is this needed?

Currently, column closures in PowerGrid only support raw HTML or simple PHP rendering logic. However, using Blade components within column closures would provide several benefits. It would enhance modularity, promote consistent UI patterns, improve readability and reusability, and seamlessly integrate with the existing Laravel ecosystem.

### Suggested Solution(s)

I propose enabling the use of Blade components directly within column closures. This would allow developers to encapsulate complex rendering logic within reusable Blade components, enhancing code maintainability and promoting consistency throughout the application.

### Related issue(s)/PR(s)

N/A

**Comment by luanfreitasdev** [comment]
Hello!, Can you give more examples and what would that affect the powergrid?. Thanks

**Reply by MGeurts** [reply]
Me too I stumbled on this issue.

I wanted to show icons in the **user->sex** column. These icons are **blade components** used all over the application.

` ->addColumn('sex', function (User $user) {
                return $user->sex == 'm' ? '<x-icon.tabler icon="gender-male" />' : '<x-icon.tabler icon="gender-female" />';
  })
`

This does not work because the `<x-icon.tabler icon="gender-male" />` is rendered as `<x-icon.tabler icon="gender-male" ></x-icon.tabler>`.

Thanks for considering.

**Comment by lsmedia** [comment]
Thumbs up for this feature request!

I would like to render a blade component with a selectbox in a table cell and would be very happy if this could be done easily.

Has anyone already found a solution or workaround for this?

**Comment by abhaypithadiya** [comment]
Hey @lsmedia!

You could do something like this:
```
            ->addColumn('status', function (OrderProduct $model) {

                $this->status[$model->id] = $model->status;

                return Blade::render(
                    <<<HTML
                <select wire:model="status.{$model->id}" wire:change="updateStatus('{$model->id}')" class="form-select cursor-pointer" style="width:auto;">
                    <option value="P">Pending</option>
                    <option value="M">In Manufacturing</option>
                    <option value="D">Delivered</option>
                    <option value="C">Cancelled</option>
                </select>
                HTML
                );
            })

```

Hope this help!

**Reply by lsmedia** [reply]
Thank you very much for this solution, @abhaypithadiya! 

I only had to make a few adjustments to my component to be able to use it in this way. Even your example with a selectbox for updating the status fits my initial situation perfectly ;)

```
return Blade::render(
   <<<HTML
      <x-input-status-select modelId="$model->id"
                             statusNum="$model->status_num"
                             onChangeCallback="update-my-model-status"
                             :candidates="MyModel::allStatus()" />
   HTML
);
```

This solution could also be suitable for @tahaRazaTalibi and @MGeurts  needs.

**Reply by abhaypithadiya** [reply]
That's great to hear! Glad the adjustments worked well for you.
